### ISSUE TITLE
Regression in 3.0.0 for Pydantic models defined using inheritance

### ISSUE THREAD

**Issue by rnathuji** [jpmorganchase/py-avro-schema#48]
We've encountered what seems like a regression in the behavior of the latest version of `py-avro-schema` when Pydantic models are defined via inheritance (e.g. to DRY up definitions where models have common fields). A minimal test case which passes in `2.2.1` but fails in `3.0.0`:

```python
def test_model_inheritance():
    class PyTypeCustomBase(pydantic.BaseModel):
        field_a: str

    class PyType(PyTypeCustomBase):
        field_b: str

    expected = {
        "type": "record",
        "name": "PyType",
        "fields": [
            {
                "name": "field_a",
                "type": "string"
            },
            {
                "name": "field_b",
                "type": "string"
            },
        ],
    }
    assert_schema(PyType, expected)
```

A potential fix may be as simple as the following :man_shrugging:  :

```patch
--- a/src/py_avro_schema/_schemas.py
+++ b/src/py_avro_schema/_schemas.py
@@ -835,7 +835,7 @@ class PydanticSchema(RecordSchema):
         # Pydantic 2 resolves forward references for us. To avoid infinite recursion, we check if the unresolved raw
         # annoation is a forward reference. If so, we use that instead of Pydantic's resolved type hint. There might be
         # a better way to un-resolve the forward reference...
-        if isinstance(self.raw_annotations[name], (str, ForwardRef)):
+        if isinstance(self.raw_annotations.get(name), (str, ForwardRef)):
```

**Comment by faph** [comment]
Cool, let me review that.

**Comment by faph** [comment]
WIP PR here: #49 

Many thanks for the reproducable test @rnathuji!

**Comment by faph** [comment]
This does highlight how fragile the forward reference "unresolving" logic is we us :cry: . For now, I am happy to accept that we are not supporting forward references in Pydantic base classes... I have added a (failing) test to make that explicit.

**Comment by faph** [comment]
@rnathuji Will let you know once merged.

**Comment by faph** [comment]
Released as version 3.0.1. Thanks @rnathuji
